auth/composite_authenticator.go

```diff
@@ -1,26 +1,18 @@
 package auth
 
-type CompositeAuthenticator struct {
-	authenticators []Authenticator
-}
-
-func NewCompositeAuthenticator(authenticators ...Authenticator) *CompositeAuthenticator {
-	return &CompositeAuthenticator{
-		authenticators: authenticators,
-	}
-}
-
-func (c *CompositeAuthenticator) Authenticate(user, pass string) (bool, error) {
-	for _, authenticator := range c.authenticators {
-		isValid, err := authenticator.Authenticate(user, pass)
-		if err != nil {
-			return false, err
-		}
+func NewCompositeAuthenticator(authenticators ...Authenticator) AuthenticatorFunc {
+	return AuthenticatorFunc(func(user, pass string) (bool, error) {
+		for _, authenticator := range authenticators {
+			isValid, err := authenticator.Authenticate(user, pass)
+			if err != nil {
```

Comment:
Just a question here. If an authenticator returns false, we continue to iterate until one authenticator returns true. Should the same apply for authenticators that return an error?

I.e., if an authenticator returns an error and it is the last authenticator return false, err otherwise continue to iterate?